### ISSUE TITLE
Expose PostTaxonomies Component for reuse in custom gutenburg blocks

### ISSUE THREAD

**Issue by kipraske** [WordPress/gutenberg#14413]
**Is your feature request related to a problem? Please describe.**
I am writing a new custom post rollup block for a project, and I thought it would be fantastic if I could just re-use the code from the [PostTaxonomies block](https://github.com/WordPress/gutenberg/tree/285875a1e2899b4999bb1396277be8a163354249/packages/editor/src/components/post-taxonomies) on my custom block. Unfortunately this block is wrapped in higher-order components when exported in order to provide the fetching initial data from the REST API and updating the post when a new category is selected. I don't think I can un-wrap these components and get the core block, and the class itself is not exported in the module so I can't access it for reuse.

**Describe the solution you'd like**
Just export the block in the module to be used as is, and create a new block for the post specific taxonomies. We can keep the one for posts as it is now as PostTaxonomies, and the new one could be TaxonomyPicker, or TaxonomyControl or something like that.

**Describe alternatives you've considered**
Right now I have a <SelectControl> with the multiple options flag turned on. This doesn't work so well when you have many tags on a website. 

I could also just copy the PostTaxonomies code and use the copy with my code. Which is what I will likely end up doing for this project as a stopgap, but I am surprised a taxonomy picker control is not available in core gutenburg. If I get to that point before anyone responds to this I will likely just start trying to implement it and send a PR your way. But I want to make sure that this is even something that the community even wants before doing that work.

**Comment by helgatheviking** [comment]
Yes! I need this exposed as a component in order to change the inputs to radio buttons in one of my plugins.

**Comment by chrisvanpatten** [comment]
I'd love to see this exposed as well. Feels like a great opportunity to standardize this common UI.

**Comment by tareiking** [comment]
Bump. This is a good example where useful UI patterns have been built, but are not extendable - therefore we don't have a standard approach for developers to follow.

**Comment by Mamaduka** [comment]
The internal components are now exported. See #38419.
